express-notes-api/fileUtils.js

```diff
@@ -0,0 +1,29 @@
+import { Buffer } from 'node:buffer';
+import { readFile, writeFile } from 'node:fs/promises';
+
+export async function getFileContent(file) {
+  try {
+    const filePath = new URL(file, import.meta.url);
+    const fileContent = await readFile(filePath, { encoding: 'utf8' });
+    return fileContent;
+  } catch (err) {
+    console.log(err);
+  }
+}
+
+export async function writeToFile(file, textContent) {
+  try {
+    if (!file) throw new TypeError('writeToFile called with undefined "file" argument');
+    if (!textContent) throw new TypeError('writeToFile called with undefined "textContent" argument');
+
+    const textToWrite = new Uint8Array(Buffer.from(textContent));
+    await writeFile(file, textToWrite);
+
+  } catch (err) {
+    console.error(err);
+  }
+}
```

Comment:
File system read / write failures during request time should result in `500 Internal Server Error` responses:

---
![image](https://github.com/jadenbertino/m2-0323-code-solutions/assets/16143771/b4472c40-5a9f-4f19-8666-e44465625bab)